`@types/bluebird-global` and `@types/jquery` are incompatible

- [x] I tried using the `@types/bluebird-global` and `@types/jquery` package and had problems.
- [x] I tried using the latest stable version of tsc. https://www.npmjs.com/package/typescript (2.9.1)
@d-ph @leonard-thieu @borisyankov

My Team has had `@types/jquery` installed and working. We are incrementally moving from babeled ES6 code to TS. All files have been moved to .ts and babel has been replaced with tsc. 

Today I needed to make Bluebird the global Promise. 
- Added window.Promise = Bluebird to main entry file 
- installed `@types/bluebird-global`

Now I am getting the following error
```
node_modules/@types/jquery/index.d.ts:6550:15 - error TS2430: Interface 'PromiseBase<TR, TJ, TN, UR, UJ, UN, VR, VJ, VN, SR, SJ, SN>' incorrectly extends interface 'Promise<TR>'.
  Types of property 'then' are incompatible.
    Type '{ <ARD = never, AJD = never, AND = never, BRD = never, BJD = never, BND = never, CRD = never, CJD...' is not assignable to type '{ <TResult1 = TR, TResult2 = never>(onfulfilled?: (value: TR) => TResult1 | PromiseLike<TResult1>...'.

6550     interface PromiseBase<TR, TJ, TN,
                   ~~~~~~~~~~~
```

If I uninstall `@types/bluebird-global` then I get tsc errors about Promise.map not existing. Hence it does appear that `@types/bluebird-global` and `@types/jquery` are not compatible, but the runtime is.

--- a/types/jquery/index.d.ts
+++ b/types/jquery/index.d.ts
@@ -6361,7 +6361,9 @@ declare namespace JQuery {
         };
 
         // Writable properties on XMLHttpRequest
-        interface XHRFields extends Partial<Pick<XMLHttpRequest, 'onreadystatechange' | 'responseType' | 'timeout' | 'withCredentials' | 'msCaching'>> { }
+        interface XHRFields extends Partial<Pick<XMLHttpRequest, 'onreadystatechange' | 'responseType' | 'timeout' | 'withCredentials'>> {
+            msCaching?: string;
+        }
     }
 
     interface Transport {

--- a/types/jquery/index.d.ts
+++ b/types/jquery/index.d.ts
@@ -37,8 +37,6 @@ declare const $: JQueryStatic;
 
 // Used by JQuery.Event
 type _Event = Event;
-// Used by JQuery.Promise3 and JQuery.Promise
-type _Promise<T> = Promise<T>;
 
 interface JQueryStatic<TElement extends Node = HTMLElement> {
     /**
@@ -6525,6 +6523,28 @@ declare namespace JQuery {
      */
     interface Thenable<T> extends PromiseLike<T> { }
 
+    // NOTE: This is a private copy of the global Promise interface. It is used by JQuery.PromiseBase to indicate compatibility with other Promise implementations.
+    //       The global Promise interface cannot be used directly as it may be modified, as in the case of @types/bluebird-global.
+    /**
+     * Represents the completion of an asynchronous operation
+     */
+    interface _Promise<T> {
+        /**
+         * Attaches callbacks for the resolution and/or rejection of the Promise.
+         * @param onfulfilled The callback to execute when the Promise is resolved.
+         * @param onrejected The callback to execute when the Promise is rejected.
+         * @returns A Promise for the completion of which ever callback is executed.
+         */
+        then<TResult1 = T, TResult2 = never>(onfulfilled?: ((value: T) => TResult1 | PromiseLike<TResult1>) | null,
+                                             onrejected?: ((reason: any) => TResult2 | PromiseLike<TResult2>) | null): JQuery._Promise<TResult1 | TResult2>;
+        /**
+         * Attaches a callback for only the rejection of the Promise.
+         * @param onrejected The callback to execute when the Promise is rejected.
+         * @returns A Promise for the completion of the callback.
+         */
+        catch<TResult = never>(onrejected?: ((reason: any) => TResult | PromiseLike<TResult>) | null): JQuery._Promise<T | TResult>;
+    }
+
     // Type parameter guide
     // --------------------
     // Each type parameter represents a parameter in one of the three possible callbacks.
@@ -6550,7 +6570,7 @@ declare namespace JQuery {
     interface PromiseBase<TR, TJ, TN,
         UR, UJ, UN,
         VR, VJ, VN,
-        SR, SJ, SN> extends _Promise<TR>, PromiseLike<TR> {
+        SR, SJ, SN> extends JQuery._Promise<TR>, PromiseLike<TR> {
         /**
          * Add handlers to be called when the Deferred object is either resolved or rejected.
          *

--- a/types/jquery/jquery-tests.ts
+++ b/types/jquery/jquery-tests.ts
@@ -6857,7 +6857,7 @@ function JQuery_jqXHR() {
         }
     }
 
-    function compatibleWithPromise(): Promise<any> {
+    function compatibleWithPromise(): JQuery._Promise<any> {
         return p;
     }
 
@@ -7279,7 +7279,7 @@ function JQuery_Promise3() {
         return s;
     }
 
-    function compatibleWithPromise(): Promise<any> {
+    function compatibleWithPromise(): JQuery._Promise<any> {
         return p;
     }
 
@@ -7423,7 +7423,7 @@ function JQuery_Promise2(p: JQuery.Promise2<string, Error, number, JQuery, strin
         return s;
     }
 
-    function compatibleWithPromise(): Promise<any> {
+    function compatibleWithPromise(): JQuery._Promise<any> {
         return p;
     }
 
@@ -7544,7 +7544,7 @@ function JQuery_Promise(p: JQuery.Promise<string, Error, number>) {
         return s;
     }
 
-    function compatibleWithPromise(): Promise<any> {
+    function compatibleWithPromise(): JQuery._Promise<any> {
         return p;
     }
 }

--- a/types/jquery/test/bluebird-global-tests.ts
+++ b/types/jquery/test/bluebird-global-tests.ts
@@ -1,0 +1,4 @@
+/// <reference types="bluebird-global" />
+
+// Pulls in bluebird-global to test compatibility.
+// Fixes https://github.com/DefinitelyTyped/DefinitelyTyped/issues/26328.

--- a/types/jquery/tsconfig.json
+++ b/types/jquery/tsconfig.json
@@ -21,6 +21,7 @@
     "files": [
         "index.d.ts",
         "jquery-tests.ts",
+        "test/bluebird-global-tests.ts",
         "test/example-tests.ts",
         "test/longdesc-tests.ts",
         "test/learn-tests.ts",

--- a/types/jquery/tslint.json
+++ b/types/jquery/tslint.json
@@ -14,6 +14,7 @@
         "no-empty-interface": false,
         "no-misused-new": false,
         "no-object-literal-type-assertion": false,
+        "no-redundant-jsdoc-2": false,
         "no-unnecessary-generics": false,
         "no-unnecessary-qualifier": false,
         "no-unnecessary-type-assertion": false,

